### PR TITLE
hooks: add uvicorn

### PR DESCRIPTION
**Pull request by imba-tjd**
I have seen https://github.com/pyinstaller/pyinstaller/pull/4664
But I think it's OK to include the whole uvicorn, because it's pure Python.

### CODE REVIEW

File: src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-uvicorn.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2020 PyInstaller Development Team.
```
Comment (rokm) [inline src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-uvicorn.py:2]: 2021

Comment (imba-tjd) [reply src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-uvicorn.py]: done

**Comment by bwoodsend** [comment]
Ughh I hate libraries that do this. Having both dynamical imports and optional dependencies means that we simultaneously must and must not use `collect_submodules()`. And our tests will always be either hopelessly inadequate in that they provide almost zero coverage or have to brute force iterate through all the permutations and take hours like our `scikit-image` tests.

**Comment by bwoodsend** [comment]
Thanks @imba-tjd.

**Pull request merged**